checks/docker/leaked_secrets.rego

```diff
@@ -0,0 +1,192 @@
+# METADATA
+# title: Secrets passed via `build-args` or envs or copied secret files
+# description: Passing secrets via `build-args` or envs or copying secret files can leak them out
+# schemas:
+# - input: schema["dockerfile"]
+# related_resources:
+# - https://docs.docker.com/build/building/secrets/
+# custom:
+#   id: DS031
+#   avd_id: AVD-DS-0031
+#   severity: CRITICAL
+#   short_code: do-not-pass-secrets
+#   recommended_action: Use secret mount if secrets are needed during image build. Use volume mount if secret files are needed during container runtime.
+#   input:
+#     selector:
+#     - type: dockerfile
+package builtin.dockerfile.DS031
+
+import rego.v1
+
+import data.lib.docker
+
+final_stage := last(input.Stages)
```

Comment:
Can this list be assumed to be always sorted?

Reply:
Dockerfile is parsed from top to bottom, so the order of stages is preserved. The order of arrays is also preserved when marshaling json (export to Rego).